### ISSUE TITLE
Checkbox not accepting input

### ISSUE THREAD

**Issue by IzzySoft** [M66B/XPrivacy#2289]
> There are four possibilities for the restriction / on demand checkboxes:
	1. &#9744; &#9744; You will not receive an on demand popup, the permission will always be allowed
	2. &#9744; [?] You will receive an on demand popup, if this times out or the screen is locked the permission will be allowed once
	3. &#9745; [?] You will receive an on demand popup, if this times out or the screen is locked the permission will be denied once
	4. &#9745; &#9744; You will not receive an on demand popup, the permission will always be denied

I cannot toggle the left checkbox, regardless where I tap. I've set XPrivacy to "on demand". Before that, the second checkbox didn't even show up, and I could only enter permanent restrictions – which is by intention, I understand; but now I can only set "on-demand" and *no* permanent restriction (i.e. options 3 and 4 are inaccessible to me, leaving only permanent grant by unchecking the "on-demand" checkbox), which I doubt is "by intention".

As I'm new to XPrivacy, it might well be me overlooking something – so I gladly accept advice in that direction as well.

Device: Motorola Milestone 2, Android 4.4.4/CM11, latest XPosed stable, latest XPrivacy. Let me know if additional information is needed – and thanks in advance!

**Comment by M66B** [comment]
Please ask questions in the XDA XPrivacy forum.
http://forum.xda-developers.com/xposed/modules/xprivacy-ultimate-android-privacy-app-t2320783

Please read this as well:
https://github.com/M66B/XPrivacy/#support

**Comment by IzzySoft** [comment]
OK, done that (and subscribed to the thread to not miss any answer). But reading > 1000 pages there is not really a good choice:

> Moreover, the answers given might be beneficial to more people than you alone.

I cannot agree more. A search in the issues would make things much easier to find, as the corresponding title helps a lot as well. Of course, it's your "baby" – so it's your way that counts :innocent:

**Comment by IzzySoft** [comment]
@M66B coming back here as I figured it, and it seems to be a minor bug: I had to disable and re-enable "on demand control" for one app to make the left-hand checkbox accepting input. After having done this for that one app, the "left boxes" worked for other apps as well – so I had not to repeat the step for each app.

Now I will "get a feeling" for the app I wanted to try out for quite a while. Put myself a pointer to support it with [Adebar](https://github.com/IzzySoft/Adebar) when generating backup scripts (trigger an export and pull the file).

**Comment by Gitoffthelawn** [comment]
I have noticed this issue as well.  I was not sure if it was a lack of understanding on my part, or a bug in XPrivacy.

**Comment by IzzySoft** [comment]
I've played a little more with it, so I might provide some additional input:

* for apps where "on-demand" was not enabled before, the issue did not happen when enabling it for them
* not sure why, but for some apps it was enabled. For those, disabling and re-enabling was required (these might have been apps installed after globally enabling on-demand, but before applying a device-reboot)
* apps installed after having enabled it globally and rebooting the device seem not to be affected

What I'm not 100% sure about is whether there might have been the "first reboot after enabling it globally" only before item number 3 – so the bug might be "worked around" by rebooting right after "globally enabling on-demand" *and before installing/updating anything new* (or, in other words, the bug might be "this happens when just enabling it globally and then trying to use it directly without a reboot in between". Can you remember anything towards that, @Gitoffthelawn ?

**Comment by Gitoffthelawn** [comment]
@IzzySoft What I've seen is that sometimes the checkboxes are non-responsive.  I've experienced it on multiple devices, both with expert mode enabled and with it disabled.  Friends have contacted me regarding the issue as well.

I remember that on one device I was able to enable *On Demand* for the entire category, and by doing so, all the checkboxes for that category became responsive.

Because I was not sure if it was a bug or just a lack of understanding on my part, I didn't pursue it, nor did I try to correlate it with any sequence of events.  Now that I know it's a bug, I'll be more observant as to what sequence of events may cause it.

**Comment by M66B** [comment]
I don't think this is a bug: to use function restrictions, you need to restrict the category first, since the function restrictions are exceptions to the category restriction.

**Comment by Gitoffthelawn** [comment]
Perhaps it is a UI design issue.  When a category is expanded, selecting the checkbox next to a single function to restrict it seems like a standard task.  I don't think making it depend on whether or not the category is restricted is in line with user-expectations.

**Comment by IzzySoft** [comment]
> to use function restrictions, you need to restrict the category first, since the function restrictions are exceptions to the category restriction.

I'm not 100% sure whether I tried toggling it for a category (but 95% sure I did). If it's not supposed to be toggled for some reason, maybe a toast should point out that reason to not leave the user "confused in the dark"?

(btw: thanks for the key. XPrivacy support has just been integrated to [Adebar](https://github.com/IzzySoft/Adebar) (devel branch, testers welcome) – offering to trigger & pull an export (XPrivacy PRO key needed) and, for root users, to pull the databases for backup purposes)
